cmake/CPM.cmake

```diff
@@ -353,19 +354,32 @@ function(cpm_parse_add_package_single_arg arg outArgs)
     elseif(scheme STREQUAL "bb")
       set(out "BITBUCKET_REPOSITORY;${uri}")
       set(packageType "git")
+    elseif(DEFINED CPM_CUSTOM_PREFIXES)
+      foreach(prefix ${CPM_CUSTOM_PREFIXES})
+        if("${prefix}" MATCHES "([^:]+):(.+)")
+          if(scheme STREQUAL "${CMAKE_MATCH_1}")
+            set(out "GIT_REPOSITORY;${uri};CUSTOM_REPOSITORY;${CMAKE_MATCH_2}")
```

Comment:
why introduce `CUSTOM_REPOSITORY` at all? It seems somewhat redundant to me

Why not concat the string here: `"GIT_REPOSITORY;${CMAKE_MATCH_2}/${uri}"`?

Reply:
Because I didn't knew it better. This was the first time I fiddled with CMake functions.